ui/pages/confirmations/components/simulation-details/useBalanceChanges.ts

```diff
@@ -102,58 +117,43 @@ function getTokenBalanceChanges(
       asset.standard === TokenStandard.ERC20 ? erc20Decimals[asset.address] : 0;
     const amount = getAssetAmount(tokenBc, decimals);
 
-    const fiatRate = tokenFiatRates[tokenBc.address];
-    const fiatAmount = fiatRate
-      ? amount.times(fiatRate).toNumber()
-      : FIAT_UNAVAILABLE;
+    const fiatRate = erc20FiatRates[tokenBc.address];
+    const fiatAmount =
+      fiatRate === undefined
+        ? FIAT_UNAVAILABLE
+        : amount.times(fiatRate).toNumber();
 
     return { asset, amount, fiatAmount };
   });
 }
 
-// Get the exchange rates for converting tokens to the user's fiat currency.
-const getTokenToFiatConversionRates = createSelector(
-  getConversionRate,
-  getTokenExchangeRates,
-  getConfirmationExchangeRates,
-  (nativeToFiat, contractExchangeRates, confirmationExchangeRates) => {
-    const mergedRates = {
-      ...contractExchangeRates,
-      ...confirmationExchangeRates,
-    } as Record<string, number>;
-    return Object.entries(mergedRates).reduce((acc, [key, value]) => {
-      acc[key.toLowerCase()] = nativeToFiat * value;
-      return acc;
-    }, {} as Record<string, number>);
-  },
-);
-
 // Compiles a list of balance changes from simulation data
 export const useBalanceChanges = (
-  simulationData?: Partial<SimulationData>,
+  simulationData: SimulationData | undefined,
 ): { pending: boolean; value: BalanceChange[] } => {
+  const chainId = useSelector(getCurrentChainId);
+  const fiatCurrency = useSelector(getCurrentCurrency);
   const nativeFiatRate = useSelector(getConversionRate);
-  const tokenFiatRates = useSelector(getTokenToFiatConversionRates);
 
-  const {
-    nativeBalanceChange,
-    tokenBalanceChanges = EMPTY_TOKEN_BALANCE_CHANGES,
-  } = simulationData ?? {};
+  const { nativeBalanceChange, tokenBalanceChanges = [] } =
+    simulationData ?? {};
 
-  const erc20Addresses = tokenBalanceChanges
+  const erc20TokenAddresses = tokenBalanceChanges
     .filter((tbc) => tbc.standard === SimulationTokenStandard.erc20)
-    .map((tbc) => tbc.address.toLowerCase() as Hex);
+    .map((tbc) => tbc.address);
 
-  const erc20Decimals = useAsyncResultOrThrow(() => {
-    return fetchErc20Decimals(erc20Addresses);
-  }, [JSON.stringify(erc20Addresses)]);
+  const erc20Decimals = useAsyncResultOrThrow(
+    () => fetchErc20Decimals(erc20TokenAddresses),
+    [JSON.stringify(erc20TokenAddresses)],
+  );
 
-  if (erc20Decimals.pending) {
-    return { pending: true, value: [] };
-  }
+  const erc20FiatRates = useAsyncResultOrThrow(
+    () => fetchTokenFiatRates(fiatCurrency, erc20TokenAddresses, chainId),
+    [JSON.stringify(erc20TokenAddresses), fiatCurrency],
```

Comment:
I guess it's unlikely to change, but should we include `chainId` in the dependencies for clarity and best practice?

Reply:
Good catch.